ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -143,16 +159,20 @@ S3Path(const std::string &dir, const std::string &name)
 static int
 S3Exist(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name, bool *exist)
 {
-    S3_STORAGE *s3;
-    s3 = FS2S3(fileSystem);
     S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
+    int ret = 0;
 
-    /* It's not in the cache, try the S3 bucket. */
+    /* Check if file in cache. */
     *exist = S3CacheExists(fileSystem, name);
-    if (!*exist)
-        return (fs->connection->ObjectExists(name, *exist));
+    if (*exist)
+        return (ret);
 
-    return (0);
+    /* It's not in the cache, try the S3 bucket. */
+    if (ret = fs->connection->ObjectExists(name, *exist) != 0)
+        return (ret);
+    FS2S3(fileSystem)->statistics->objectExistsCount++;
```

Comment:
Do we want this statistic to be incremented even if S3Connection::ObjectExists fails?

Reply:
I currently have it so that the S3 operation counts only increment after successful calls. I don't think it makes too much of a difference, as the count will only differ by one, since WiredTiger will just error and fail if the operation fails.

Reply:
Technically what might happen is WiredTiger will retry on a failure. Failures should be rare and are better captured in the logs. Statistically, they are expected to be rare enough that it does not make a lot of sense to have a running statistic on them. I am okay to collect only successful calls to S3 for now.